### ISSUE TITLE
Bro script should escape headers when writing to file

### ISSUE THREAD

**Issue by davidwagner** [mikeaboody/phishing-research#18]
When writing header names and values, the Bro script should escape all data properly before writing to disk, otherwise we'll have errors when you try to parse them (with eval()).

A plausible algorithm:
1. Escape all backslashes: replace `\` with `\\` or `\x5C` (the latter might be slightly more robust)
2. Escape all quotes: replace `"` with `\"` or `\x22` (the latter might be slightly more robust)
3. Potentially: Escape all characters outside the 0x00-0x7F set: e.g., replace with a hex escape (I'm not sure whether this is necessary; it will depend on what charset encodings Bro and Python use, and how Bro and Python handle characters outside the 0x00-0x7F set)

Order matters: doing 2 then 1 will obviously give the wrong result.

Both header names and values should be escaped.

**Comment by nexusapoorvacus** [comment]
80cc61938131d6bebca8420108c5792ea97593de

**Comment by davidwagner** [comment]
Did you test this?   If I understand how string literals in Bro work, the sequence of 3 gsub() calls look like they are going to have no effect: you replace `'` with `'` (I think `"\'"` might be equivalent to a string containing one character, the single quote), and so on.

(Alternatively, if I've misunderstood how Bro interprets backslashes in string literals, it looks the gsub() calls are in the wrong order, so I suspect this isn't going to work right: the effect of the sequence of 3 gsub()s will be to replace each `'` with `\\'` (i.e., one too many backslashes).)

Would Bro's to_string_literal() be a better solution than gsub()+escape_string()?  Seems like you might be able to replace all 4 lines (3 gsub()s + 1 escape_string()) with a single call to to_string_literal().  https://www.bro.org/sphinx/scripts/base/bif/strings.bif.bro.html#id-to_string_literal

**Comment by nexusapoorvacus** [comment]
I think to_string_literal() is much better than what I was doing before. I have pushed the new main.bro to master. dc08ed92407564ed1e4af86d39579d29e82d6c46

I was not able to test it very well since Bro is not able to detect any emails on the new pcaps I generate using tcpdump on the mail server. I need to generate these pcaps so that I can test the behavior when a header value has a quote or backslash. I am not sure what the problem is since the pcaps I generated the same way month ago are detected by Bro. I am still trying to debug this issue.
